apps/frontend/src/pages/points-of-interest-details/libs/components/route-gallery/route-gallery.tsx

```diff
@@ -0,0 +1,50 @@
+import { RouteCard } from "~/libs/components/route-card/route-card.js";
+import { sortByDate } from "~/libs/helpers/helpers.js";
+import { useMemo } from "~/libs/hooks/hooks.js";
+import { type PointsOfInterestGetByIdResponseDto } from "~/modules/points-of-interest/points-of-interest.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	routes: PointsOfInterestGetByIdResponseDto["routes"];
+};
+
+const HALF = 2;
+
+const RoutesGallery = ({ routes }: Properties): React.JSX.Element => {
+	const cards = useMemo(
+		() =>
+			routes.map(({ geometry, id, images, name, pois }) => {
+				const cardImage = sortByDate(images, "createdAt").at(0)?.url;
+				const coordsCenter = Math.floor(geometry.coordinates.length / HALF);
+				const routeCenter = geometry.coordinates[coordsCenter] as [
+					number,
+					number,
+				];
+
```

Comment:
```suggestion
const HALF = 2;

const RoutesGallery = ({ routes }: Properties): React.JSX.Element => {
	const cards = useMemo(
		() =>
			routes.map(({ geometry, id, images, name, pois }) => {
				const cardImage = sortByDate(images, "createdAt").at(0)?.url;
				const coordsCenter = Math.floor(geometry.coordinates.length / HALF);
				const routeCenter = geometry.coordinates[coordsCenter] as [
					number,
					number,
				];

```
We already have fitBounds in MapProvider, but it's in main, so if you merge it, all you need to do is pass routeline and markers to Map

Reply:
Other things look good to me